x-pack/plugins/code/server/lsp/java_launcher.ts

```diff
@@ -115,7 +116,11 @@ export class JavaLauncher implements ILanguageServerLauncher {
 
     if (this.getSystemJavaHome()) {
       const javaHomePath = this.getSystemJavaHome();
-      if (await this.checkJavaVersion(javaHomePath)) {
+      const javaVersion = await this.getJavaVersion(javaHomePath);
+      if (javaVersion == 8) {
+        this.needExtraOptions = false;
+      }
+      if (javaVersion >= 8) {
```

Comment:
do you mean <= 8, we do false, and > 8 we do true?

Reply:
Yes

Reply:
then shouldn't we do?

```
if (javaVersion <= 8) {
        this.needModuleArguments = false;
      }
if (javaVersion > 8) {
``
?

Reply:
Maybe the semantic here is not very clear. If java version < 8, we use the bundled JDK(whose version >8), so we need extra arguments by default. When system jdk's version >= 8, we use the system JDK, while only for jdk's version = 8, we needn't extra arguments.

Reply:
ok, maybe use <= instead of ==, although it cover more case that not existed, yet it's easier to understand.

or you could put those comments there so people don't get confuse